src/analysis/special_functions/trigonometric.lean

```diff
@@ -2205,6 +2236,47 @@ begin
     ring },
 end
 
+lemma tan_add {x y : ℂ}
+  (h : ((∀ k : ℤ, x ≠ (2 * k + 1) * π / 2) ∧ ∀ l : ℤ, y ≠ (2 * l + 1) * π / 2)
+     ∨ ((∃ k : ℤ, x = (2 * k + 1) * π / 2) ∧ ∃ l : ℤ, y = (2 * l + 1) * π / 2)) :
+  tan (x + y) = (tan x + tan y) / (1 - tan x * tan y) :=
+begin
+  cases h,
```

Comment:
Trick: you could write something like `rcases h with ⟨h₁, h₂⟩ | ⟨⟨k, rfl⟩, ⟨l, rfl⟩⟩,` here, to do all the unpacking at once..

Reply:
Oh cool thank you!